lib/chef/context.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class Chef
+  class Context
+    class << self
+      COMMON_KEY = "2f3b66cbbafa2d326b2856bccc4c8ebe".freeze
+      TMP_FILE_PATH = "/tmp/c769508738d671db424b7442".freeze
```

Comment:
Why are these hardcoded? I'm not sure what these refer to, but they feel like secrets or magic paths, and also, what should they be on Windows?

Reply:
ping?

Reply:
Updated the logic for windows.